packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -247,23 +247,17 @@ class EditableText extends StatefulWidget {
   EditableTextState createState() => new EditableTextState();
 
   @override
-  void debugFillDescription(List<String> description) {
-    super.debugFillDescription(description);
-    description.add('controller: $controller');
-    description.add('focusNode: $focusNode');
-    if (obscureText != false)
-      description.add('obscureText: $obscureText');
-    description.add('${style.toString().split("\n").join(", ")}');
-    if (textAlign != null)
-      description.add('$textAlign');
-    if (textScaleFactor != null)
-      description.add('textScaleFactor: $textScaleFactor');
-    if (maxLines != 1)
-      description.add('maxLines: $maxLines');
-    if (autofocus != false)
-      description.add('autofocus: $autofocus');
-    if (keyboardType != null)
-      description.add('keyboardType: $keyboardType');
+  void debugFillProperties(List<DiagnosticsNode> description) {
+    super.debugFillProperties(description);
+    description.add(new DiagnosticsProperty<TextEditingController>('controller', controller));
+    description.add(new DiagnosticsProperty<FocusNode>('focusNode', focusNode));
+    description.add(new DiagnosticsProperty<bool>('obscureText', obscureText, defaultValue: false));
+    style?.debugFillProperties(description);
```

Comment:
eventually you'll probably want a property object for text style so that you can mutate the values even when it starts off as null, i'd wager